tile/TileSeparateCrossbarRTL.py

```diff
@@ -186,23 +186,44 @@ def update_opt():
       s.routing_crossbar.recv_opt.msg @= s.ctrl_mem.send_ctrl.msg
       s.fu_crossbar.recv_opt.msg @= s.ctrl_mem.send_ctrl.msg
 
-      s.element.recv_opt.en @= s.ctrl_mem.send_ctrl.en
-      s.routing_crossbar.recv_opt.en @= s.ctrl_mem.send_ctrl.en
-      s.fu_crossbar.recv_opt.en @= s.ctrl_mem.send_ctrl.en
-
-      s.ctrl_mem.send_ctrl.rdy @= s.element.recv_opt.rdy & \
-                                  s.routing_crossbar.recv_opt.rdy & \
-                                  s.fu_crossbar.recv_opt.rdy
-
+      s.element.recv_opt.val @= s.ctrl_mem.send_ctrl.val & ~s.element_done
+      s.routing_crossbar.recv_opt.val @= s.ctrl_mem.send_ctrl.val & ~s.routing_crossbar_done
+      s.fu_crossbar.recv_opt.val @= s.ctrl_mem.send_ctrl.val & ~s.fu_crossbar_done
+
+      # s.ctrl_mem.send_ctrl.rdy @= s.element.recv_opt.rdy & \
+      #                             s.routing_crossbar.recv_opt.rdy & \
+      #                             s.fu_crossbar.recv_opt.rdy
+
+      # FIXME: yo96, rename ctrl.rdy to ctrl.proceed or sth similar.
+      # Allows either the FU-related go out first or routing-xbar go out first. And only
+      # allows the ctrl signal proceed till all the sub-modules done their own job (once).
+      s.ctrl_mem.send_ctrl.rdy @= (s.element.recv_opt.rdy | s.element_done) & \

```

Comment:
@yo96 The `xxx_done` is the `status` we discussed (which is updated in the following `update_ff` block). Plz let me know if you have any concerns about it. Cleanup can be in another PR though.

Reply:
Looks okay to me. Didn't see anything that is obviously wrong.